src/main/java/parser/Parser.java

```diff
@@ -57,11 +88,92 @@ public Command getUserCommand(String userInput) throws TantouException {
+     * argument is incorrectly formatted, a {@link TantouException} is thrown.</p>
+     *
+     * @param userInput a String representing the user input containing
+     *                  command-line arguments. It may include options
+     *                  and arguments in quoted format.
+     * @return an array of strings representing the parsed user input,
+     *         with each option and argument as separate entries.
+     * @throws TantouException if any option's argument is not enclosed in quotes
+     *                         or if the input format is invalid.
+     */
+    public String[] getUserInputAsList(String userInput) throws TantouException {
+        Matcher matcher = USER_COMMAND_REGEX.matcher(userInput);
+        ArrayList<String> list = new ArrayList<>();
+
+        while (matcher.find()) {
+            String match = matcher.group();
+            list.add(match);
+        }
+
+        validateArguments(list);
+
+        return list.toArray(new String[0]);
+    }
+
+    /**
+     * Validates the arguments provided in the list to ensure that
+     * all options have their corresponding arguments enclosed in quotes.
+     *
+     * <p>This method iterates through the list of arguments and checks each
+     * option against the predefined options in the {@link constants.Options#OPTIONS_ARRAY}.
+     * If an option is found, it verifies that the next element in the list is a
+     * quoted string. If the argument is not quoted, a {@link TantouException}
+     * is thrown with a descriptive message.</p>
+     *
+     * @param list an ArrayList of Strings representing the command-line arguments
+     *             provided by the user. Each option must have its argument in
+     *             quotes (e.g., "-a \"Kubo Tite\"").
+     * @throws TantouException if any option's argument is not enclosed in quotes.
+     */
+    public void validateArguments(ArrayList<String> list) throws TantouException {
+        for (int i = 0; i < list.size(); i++) {
+            for (String [] option : OPTIONS_ARRAY) {
+                String optionWithDash = "-" + option[SHORT_OPTION_INDEX];
+                validateArgument(list, optionWithDash, i);
+            }
+        }
+    }
+
+    /**
+     * Validates that the argument following an option is a quoted string.
+     *
+     * @param list the command-line arguments
+     * @param optionIndex the index of the option being validated
+     * @throws TantouException if the argument is not enclosed in quotes
+     */
+    public void validateArgument(ArrayList<String> list, String optionWithDash, int optionIndex)
+            throws TantouException {
+        if (optionWithDash.equals(list.get(optionIndex))) {
+            // Check if the next element is a quoted string
+            if (optionIndex + 1 >= list.size() || !isQuotedString(list.get(optionIndex + 1))) {
+                throw new TantouException("Arguments must be in quotes, e.g., -a \"Kubo Tite\"");
```

Comment:
I like that we have custom exceptions for our app, but I do feel that we can go further to create even more sub-exceptions (e.g. `InvalidCommandException`) so we can be more expressive without defaulting to writing different error messages when creating a new `TantouException`.